usr/share/rear/finalize/GNU/Linux/240_reassign_luks_keyfiles.sh

```diff
@@ -0,0 +1,45 @@
+# Re-assign original keyfiles to LUKS volumes
+#
+# In the 'layout/prepare' stage, temporary keyfiles were generated for password-less decryption. By now, the
+# original keyfiles should have been restored from the backup. If so, the original keyfiles are re-assigned
+# to their LUKS volumes and temporary keyfiles are discarded. Where an original keyfile was not restored
+# to its expected location, an error message is displayed and the corresponding temporary keyfile will take
+# over, so that the recovered system remains fully functional.
+
+local device original_keyfile
+
+awk '
+    $1 == "crypt" && / keyfile=/ {
+        device = $3;
+
+        sub("^.* keyfile=", "");
+        sub("[ \t].*$", "");
+        original_keyfile = $0;
+
+        print device, original_keyfile;
+    }
+' "$LAYOUT_FILE" |
+while read device original_keyfile; do
+    Log "Re-assigning keyfile $original_keyfile to LUKS device $device"
+
+    # The scheme for generating a temporary keyfile path must be the same here and in the 'layout/prepare' stage.
+    temp_keyfile="${TMPDIR:-/tmp}/LUKS-keyfile-$(basename $original_keyfile)"
```

Comment:
What if I've used following keys for encryption:
```
/secure_dir1/key1
/secure_dir2/key1
```

Maybe I've misunderstood something, but wouldn't this mean that disks encryption will be done by same key, when in reality we've used two keys?

Reply:
Yes. Good catch. Will have to correct this. I happen to use a configuration where key files are kept in the same directoriy.